atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1427,9 +1430,124 @@ public Void apply(Client client) throws Exception {
+     * This is required because Cassandra CAS does not treat setting to null / empty as a delete,
+     * (though there is a to-do in their code to possibly add this)
+     * though it accepts an empty expected column to mean that non-existance was expected
+     * (see putUnlessExists for example, which has the luck of not having to ever deal with deleted values)
+     *
+     * I can't hold this against them though, because Atlas has a semi-similar problem with empty byte[] values meaning deleted internally.
+     *
+     * @return an ID to be passed into a subsequent unlock call
+     */
+    public long waitForSchemaMutationLock() {
+        final long perOperationNodeIdentifier = ThreadLocalRandom.current().nextLong(Long.MAX_VALUE - 2);
+
+        try {
+            clientPool.runWithRetry(new FunctionCheckedException<Client, Void, Exception>() {
+                @Override
+                public Void apply(Client client) throws Exception {
+                    Cell globalDdlLockCell = Cell.create(CassandraConstants.GLOBAL_DDL_LOCK.getBytes(), CassandraConstants.GLOBAL_DDL_LOCK_COLUMN_NAME.getBytes());
+                    ByteBuffer rowName = ByteBuffer.wrap(globalDdlLockCell.getRowName());
+                    Column ourUpdate = lockColumnWithValue(Longs.toByteArray(perOperationNodeIdentifier));
+
+                    List<Column> expected = ImmutableList.of(lockColumnWithValue(Longs.toByteArray(CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE)));
+
+                    CASResult casResult = writeLockWithCAS(client, rowName, expected, ourUpdate);
+
+                    int timesAttempted = 0;
+                    long timeSlept = 0;
+
+                    while (!casResult.isSuccess()) { // could have a timeout controlling this level, confusing for users to set both timeouts though
+                        if (casResult.getCurrent_valuesSize() == 0) { // never has been an existing lock
+                            // special case, no one has ever made a lock ever before
+                            // this becomes analogous to putUnlessExists now
+                            expected = ImmutableList.<Column>of();
+                        } else {
+                            Column existingValue = Iterables.getOnlyElement(casResult.getCurrent_values(), null);
+                            if (existingValue == null) {
+                                throw new IllegalStateException("Something is wrong with underlying locks. Consult support for guidance on manually examining and clearing locks from " + CassandraConstants.LOCK_TABLE + " table.");
+                            }
+                            expected = ImmutableList.of(lockColumnWithValue(Longs.toByteArray(CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE)));
+                        }
+
+                        if (timeSlept > configManager.getConfig().schemaMutationTimeoutMillis() * 4) { // possibly dead remote locker
```

Comment:
(background for other people reading, I originally had automated logic here to after a few minutes of looking at the same lock resume everything assuming the existing lock was dead)
I understand but personally dislike this balance of safety vs. likelihood of support burden / manual fixup, because I think this case is going to happen often (cassandra table creation / schema gossip waiting is a slow and occasionally error prone process)

Maybe as an in-between, this error could at least include what we would always have people do, which is shut down all atlases and cqlsh truncate _locks